addons/website_event/views/event_templates.xml

```diff
@@ -11,710 +11,775 @@
+                </t>
+            </div>
+        </li>
     </xpath>
 </template>
 
-<template id="event_left_column" inherit_id="website_event.index" customize_show="True" name="Filters Column on the Left">
-    <xpath expr="//div[@id='middle_column']" position="before">
-        <div class="col-lg-3 css_noprint" id="left_column"/>
+<!-- Filter - Location -->
+<template id="event_location" inherit_id="website_event.index_topbar" active="True" customize_show="True" name="Filter by Country">
+    <xpath expr="//ul[hasclass('o_wevent_index_topbar_filters')]" position="inside">
+        <li class="nav-item dropdown mr-2 my-1 my-lg-0">
+            <a href="#" role="button" class="dropdown-toggle" data-toggle="dropdown">
+                <i class="fa fa-map-marker"/>
+                <t t-if="current_country" t-esc="current_country.name"/>
+                <t t-else="">All countries</t>
+            </a>
+            <div class="dropdown-menu">
+                <t t-foreach="countries" t-as="country">
+                    <t t-if="country['country_id']">
+                        <a t-attf-href="/event?{{ keep_query('type', 'date', country=country['country_id'][0]) }}" t-attf-class="dropdown-item d-flex align-items-center justify-content-between #{searches.get('country') == str(country['country_id'] and country['country_id'][0]) and 'active' or ''}">
+                            <t t-esc="country['country_id'][1]"/>
+                            <span t-esc="country['country_id_count']" class="badge badge-pill ml-auto"/>
+                        </a>
+                    </t>
+                    <t t-else="">
+                        <a t-attf-href="/event?{{ keep_query('type', 'date', country='online') }}" t-attf-class="dropdown-item d-flex align-items-center #{searches.get('country') == 'online' and ' active' or ''}">
+                            <span>Online Events</span>
+                            <span t-esc="country['country_id_count']" class="badge badge-pill ml-3"/>
+                        </a>
+                    </t>
+                </t>
+            </div>
+        </li>
     </xpath>
 </template>
 
-<template id="event_time" customize_show="True" inherit_id="website_event.event_left_column" name="Filter by Date">
-    <xpath expr="//div[@id='left_column']" position="inside">
-        <ul class="nav nav-pills flex-column mb-3">
-            <t t-foreach="dates" t-as="date">
-                <li class="nav-item" t-if="date[3] or (date[0] in ('old','all'))">
-                    <a t-attf-href="/event?{{ keep_query('country', 'type', date=date[0]) }}" t-attf-class="nav-link d-flex align-items-center #{searches.get('date') == date[0] and ' active' or ''}"><t t-esc="date[1]"/>
-                        <span t-if="date[3]" class="badge badge-primary ml-auto"><t t-esc="date[3]"/></span>
-                    </a>
-                </li>
-            </t>
-        </ul>
-    </xpath>
+<!-- Search Box -->
+<!-- TODO - Make it works -->
```

Comment:
@qsm-odoo Search Box. I need you for this part. I can take care of the layout/design of the results page.